pkg/common/fflag/fflag_test.go

```diff
@@ -0,0 +1,157 @@
+package fflag
+
+import (
+	"sync"
+	"testing"
+
+	"github.com/hashicorp/hcl"
+	"github.com/hashicorp/hcl/hcl/ast"
+)
+
+const (
+	flagTestFlag Flag = "i_am_a_test_flag"
+)
+
+var (
+	supplementalTestFlag = map[Flag]bool{
+		flagTestFlag: false,
```

Comment:
nit: I think some of the test scaffolding (e.g. preSet and preUnset) could be removed and the testing simplified if we just defined two test flags; one that had a default true value, and the other a default false value.

Reply:
nice! will do